src/transations.rs

```diff
@@ -0,0 +1,1359 @@
+                [RevaultTxOut::UnvaultTxOut(ref unvault_txout), RevaultTxOut::CpfpTxOut(ref cpfp_txout)],
+            ) => {
+                let vault_input = TxIn {
+                    previous_output: *vault_prevout,
+                    ..Default::default()
+                };
+                Ok(RevaultTransaction::UnvaultTransaction(Transaction {
+                    version: 2,
+                    lock_time: 0, // FIXME: anti fee snipping
+                    input: vec![vault_input],
+                    output: vec![unvault_txout.clone(), cpfp_txout.clone()],
+                }))
+            }
+            _ => Err(RevaultError::TransactionCreation(format!(
+                "Unvault: type mismatch on prevout ({:?}) or output(s) ({:?})",
+                prevouts, txouts
+            ))),
+        }
+    }
+
+    /// Create a new spend transaction.
+    /// A spend transaction can batch multiple unvault txouts, and may have any number of
+    /// txouts (including, but not restricted to, change).
+    ///
+    /// # Errors
+    /// - If prevouts or txouts type mismatch.
+    pub fn new_spend(
+        prevouts: &[RevaultPrevout],
+        outputs: &[RevaultTxOut],
+        csv_value: u32,
+    ) -> Result<Self, RevaultError> {
+        let mut txins = Vec::<TxIn>::with_capacity(prevouts.len());
+        for prevout in prevouts {
+            if let RevaultPrevout::UnvaultPrevout(ref prev) = prevout {
+                txins.push(TxIn {
+                    previous_output: *prev,
+                    sequence: csv_value,
+                    ..TxIn::default()
+                })
+            } else {
+                return Err(RevaultError::TransactionCreation(format!(
+                    "Spend: prevout ({:?}) type mismatch",
+                    prevout
+                )));
+            }
+        }
+
+        let mut txouts = Vec::<TxOut>::with_capacity(outputs.len());
+        for out in outputs {
+            match out {
+                RevaultTxOut::SpendTxOut(ref txout) | RevaultTxOut::VaultTxOut(ref txout) => {
+                    txouts.push(txout.clone())
+                }
+                _ => {
+                    return Err(RevaultError::TransactionCreation(format!(
+                        "Spend: output ({:?}) type mismatch",
+                        out
+                    )))
+                }
+            }
+        }
+
+        Ok(RevaultTransaction::SpendTransaction(Transaction {
+            version: 2,
+            lock_time: 0,
+            input: txins,
+            output: txouts,
+        }))
+    }
+
+    /// Create a new cancel transaction.
+    /// A cancel transaction always pays to a vault output and spend the unvault output, and
+    /// may have a fee-bumping input.
+    ///
+    /// # Errors
+    /// - If prevouts or txouts type mismatch.
+    pub fn new_cancel(
+        prevouts: &[RevaultPrevout],
+        txouts: &[RevaultTxOut],
+    ) -> Result<RevaultTransaction, RevaultError> {
```

Comment:
nit: `new_unvault()` and `new_spend()` return `Result<Self, RevaultError>`, 
while `new_cancel()` and `new_emergency()` return `Result<RevaultTransaction, RevaultError>`. I don't think there's a functional difference but maybe its better to be consistent.

Reply:
Absolutely, thanks!